src/main/java/net/smartcosmos/edge/things/service/GetThingEdgeServiceDefault.java

```diff
@@ -139,8 +154,39 @@ public GetThingEdgeServiceDefault(
         return buildForwardingResponse(thingResponse);
     }
 
+    private ResponseEntity<?> getThingsMergeMetadataUnpaged(
+        String type,
+        Map<String, Set<String>> urns,
+        ResponseEntity thingResponse,
+        SmartCosmosUser user) {
+
+        if (thingResponse.hasBody() && thingResponse.getBody() instanceof RestUnpagedThingResponse) {
+
+            RestUnpagedThingResponse things = (RestUnpagedThingResponse) thingResponse.getBody();
+            try {
+                List<Map<String, Object>> data = collectFindByTypeData(things.getData(), null, user);
+
+                RestEdgeThingResponseDto<Map<String, Object>> responsePage = RestEdgeThingResponseDto.<Map<String, Object>>builder()
+                    .data(data)
+                    .build();
+
+                return ResponseEntity.ok()
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .body(responsePage);
```

Comment:
This is confusing: The method name says `Unpaged` while the variable is `responsePage`. I'm not sure `RestEdgeThingResponseDto` is the correct model, I assume it's the standard response model which doesn't fit the API specification for `findByUrns`.